data-raw/spawning-habitat-decay/spawning-decay.R

```diff
@@ -0,0 +1,413 @@
+  
+  if (w %in% watersheds_with_decay) {
+    dsm_flows |>
+      filter(watershed == w, year(date) %in% 1979:2000) |> 
+      mutate(flow_adjusted = flow_cfs - x, 
+             decay_min = MIN_flow_cfs_to_sed_cfd_final(flow_adjusted), 
+             decay_avg = AVG_flow_cfs_to_sed_cfd_final(flow_adjusted), 
+             decay_max = MAX_flow_cfs_to_sed_cfd_final(flow_adjusted) 
+      ) |> 
+      pivot_longer(names_to = "decay_type", values_to = "decay_amount", -c(date, watershed, 
+                                                                           flow_cfs, flow_adjusted)) |> 
+      mutate(decay_cfd = ifelse(is.na(decay_amount), 0, decay_amount), 
+             decay_cfm = decay_cfd * days_in_month(month(date)),
+             decay_sqm = decay_cfm / 2, 
+             decay_acres_month = decay_sqm / 43560, 
+             decay_type = stringr::str_extract(decay_type, "min|avg|max")) |> 
+      select(date, watershed, flow_cfs, decay_type, decay_acres_month) 
+  } else {
+    dsm_flows |> 
+      filter(watershed == w, year(date) %in% 1979:2000) |> 
+      transmute(
+        date, watershed, flow_cfs, decay_type = "none", decay_acres_month = 0
+      )
+  }
+}) |> 
+  set_names(fallRunDSM::watershed_labels)
+
+usethis::use_data(watershed_spawning_decays, overwrite = TRUE)
+
+watershed_decays$`American River` |> 
+  ggplot(aes(date, decay_acres_month, color = decay_type)) + geom_line()
+
+watershed_decays$`American River` |> 
+  group_by(decay_type) |> 
+  mutate(agg_decay = cumsum(-decay_acres_month)) |> 
+  ggplot(aes(date, agg_decay, color = decay_type)) + geom_line()
+
+# steps we need to go through to apply the decay on this dataset
+# 1. get the decay amount per watershed, have it filtered to the level min,avg,max,none
+# compute the amount of habitat to be removed
+watershed_decay_level_lookups <- c(
+  `Upper Sacramento River` = "min", `Antelope Creek` = "none", 
+  `Battle Creek` = "none", `Bear Creek` = "none", `Big Chico Creek` = "none", 
+  `Butte Creek` = "none", `Clear Creek` = "min", `Cottonwood Creek` = "none", 
+  `Cow Creek` = "none", `Deer Creek` = "none", `Elder Creek` = "none", 
+  `Mill Creek` = "none", `Paynes Creek` = "none", `Stony Creek` = "max", 
+  `Thomes Creek` = "none", `Upper-mid Sacramento River` = "none", 
+  `Sutter Bypass` = "none", `Bear River` = "none", `Feather River` = "avg", 
+  `Yuba River` = "min", `Lower-mid Sacramento River` = "none", `Yolo Bypass` = "none", 
+  `American River` = "max", `Lower Sacramento River` = "none", `Calaveras River` = "min", 
+  `Cosumnes River` = "none", `Mokelumne River` = "avg", `Merced River` = "avg", 
+  `Stanislaus River` = "avg", `Tuolumne River` = "min", `San Joaquin River` = "none"
+)
+
+
+watersheds_with_decay <- names(which(DSMhabitat::watershed_decay_status))
+
+spawning_decay_multiplier <- purrr::map(names(watershed_decay_level_lookups), function(w) {
+  if (w %in% watersheds_with_decay) {
+    decay <- DSMhabitat::watershed_decays[[w]] |> 
```

Comment:
same comment as above. Was not able to run this function

Reply:
fixed